front/src/components/HomePage/HomePage.js

```diff
@@ -1,24 +1,43 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import { connect } from "react-redux";
 import fetchBooks from "../../state/books/actions";
+import Search from "./Search";
 import BookCard from "./BookCard";
 import style from "./HomePage.module.css";
 
-class HomePage extends React.Component {
+const HomePage = (props) => {
+  const { fetchBooks, books } = props;
 
-  componentDidMount() {
-    this.props.fetchBooks()
-  }
+  const [filteredBooks, setFilteredBooks] = useState({
+    filteredTitles: [],
+    filter: "",
+  });
```

Comment:
```suggestion
  const [bookList, setBookList] = useState(books);
  const [searchTerm, setSearchTerm] = useState("");
```
I would split this state into `bookList` and `searchTerm`. Then on the `searchTerm` update, we can update `bookList`. It is more readable in my opinion :)